akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -986,32 +1011,36 @@ private[akka] class ShardRegion(
   def registrationMessage: Any =
     if (entityProps.isDefined) Register(self) else RegisterProxy(self)
 
-  def requestShardBufferHomes(): Unit = {
-    // Have to use vars because MessageBufferMap has no map, only foreach
-    var totalBuffered = 0
-    var shards = List.empty[String]
-    shardBuffers.foreach {
-      case (shard, buf) =>
-        coordinator.foreach { c =>
+  /**
+   * Send GetShardHome for all shards with buffered messages
+   * If coordinator is empty, nothing happens
+   */
+  def tryRequestShardBufferHomes(): Unit = {
+    coordinator.foreach { coord =>
+      // Have to use vars because MessageBufferMap has no map, only foreach
+      var totalBuffered = 0
+      var shards = List.empty[String]
+      shardBuffers.foreach {
+        case (shard, buf) =>
           totalBuffered += buf.size
           shards ::= shard
           log.debug(
-            "{}: Retry request for shard [{}] homes from coordinator at [{}]. [{}] buffered messages.",
+            "{}: Requesting shard home for [{}] to coordinator at [{}]. [{}] buffered messages.",
             typeName,
             shard,
-            c,
+            coord,
             buf.size)
-          c ! GetShardHome(shard)
-        }
-    }
+          coord ! GetShardHome(shard)
+      }
 
-    if (retryCount >= 5 && retryCount % 5 == 0 && log.isWarningEnabled) {
-      log.warning(
-        "{}: Retry request for shards [{}] homes from coordinator. [{}] total buffered messages. Coordinator [{}]",
-        typeName,
-        shards.sorted.mkString(","),
-        totalBuffered,
-        coordinator)
+      if (retryCount >= 5 && retryCount % 5 == 0 && log.isWarningEnabled) {
+        log.warning(
+          "{}: Requested shard homes [{}] to coordinator at [{}]. [{}] total buffered messages.",
+          typeName,
+          shards.sorted.mkString(","),
+          coord,
+          totalBuffered)
+      }
```

Comment:
Did some re-wording here. It think it's a little more clear now what have happened. 

See my comments starting at https://github.com/akka/akka/pull/29599#discussion_r492827690 for context for why I believe this log should be inside `coordinator.foreach` block.

Reply:
"from coordinator" rather than "to"?

Reply:
It was 'from' and I found it strange. 

I request something to you, not from you.

Reply:
hmm, it seems that I'm wrong
https://english.stackexchange.com/questions/113846/requested-to-vs-requested-from

Again

Reply:
Apology accepted

Reply:
sorry ;-)